doc/source/ray-references/api.rst

```diff
@@ -13,4 +13,4 @@ API References
     ../rllib/package_ref/index.rst
     ../workflows/package-ref.rst
     ../cluster/package-overview.rst
-    ../ray-core/package-ref.rst
+    ../ray-core/reference/index.rst
```

Comment:
Checked several libraries (pandas, spark, numpy, scipy), seems there api docs have the following structure:

```
reference/
    index.rst
    group1.rst
    group2.rst
    api|generated/class1.rst
    api|generated/class1.rst
```

I feel it's better than

```
api/
    api.rst
    group1.rst
    group2.rst
    doc/class1.rst
    doc/class1.rst
```

so I tried it out in this PR. Would like to get other's opinions on this.